Document/0x06f-Testing-Local-Authentication.md

```diff
@@ -193,10 +193,14 @@ If `Security.framework` is used, only the second one will be shown.
 
 #### Static Analysis
 
-It is important to remember that Local Authentication framework is an event-based procedure and as such, should not the sole method of authentication. Though this type of authentication is effective on the user-interface level, it is easily bypassed through patching or instrumentation.
+It is important to remember that Local Authentication framework is an event-based procedure and as such, should not be the sole method of authentication. Though this type of authentication is effective on the user-interface level, it is easily bypassed through patching or instrumentation. Therefore, it is best to use the Keychain service method. Which means you should:
 
 - Verify that sensitive processes, such as re-authenticating a user triggering a payment transaction, are protected using the Keychain services method.
-- Verify that the `kSecAccessControlTouchIDAny` or `kSecAccessControlTouchIDCurrentSet` flags are set and `kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly` protection classes are set when the `SecAccessControlCreateWithFlags` method is called. Note that, alternatively, `kSecAccessControlUserPresence` can be used as a flag as well when you want to be able to use passcode as a fallback. Last, note that, when `kSecAccessControlTouchIDCurrentSet` is set, changing the fingerprints registered to the device will invalidate the entry which is protected with the flag.
+- Verify that access control flags are set for the keychain entry which ensure that the keychain item its data can only be unlocked by means of authenticating the user. This can be with one of the following flags:
+  - `kSecAccessControlTouchIDCurrentSet`(since iOS 11.3 `kSecAccessControlBiometryCurrentSet`). This will make sure that a user needs to authenticate with his biometrics (e.g. Face ID or Touch ID)before he can access the data in the Keychain entry. Whenever the user adds a fingerprint or facial representation to the device, it will automatically invalidate the entry in the Keychain. This makes sure that the Keychain-entry is protected against adversarial additions to the Keychain.
+  - `kSecAccessControlTouchIDAny` (since iOS 11.3 `kSecAccessControlBiometryAny`). This will make sure that a user needs to authenticate with his biometrics (e.g. Face ID or Touch ID)before he can access the data in the Keychain entry. The Keychain-entry will survive any (re-)enroling of new fingerprints or facial representation. This can be very practical if the user has a changing fingerprint (which happens in case of lumberjackers for instance). However, it also means that attackers, who are somehow able to enrole their fingerprints or facial representations to the device, can now access those entries as well.
+  - `kSecAccessControlUserPresence` can be used as an alternative. This will allow the user to authenticate himself through a PassCode if the biometric authentication no longer works.
```

Comment:
```suggestion
  - `kSecAccessControlUserPresence` can be used as an alternative. This will allow the user to authenticate himself through a PassCode if the biometric authentication no longer works. This is considered to be weaker than kSecAccessControlBiometryAny since it is much easier to steal someone's pincode by seeing them enter it, than it is to bypass the Touch ID or Face ID service.
```